### DISCUSSION TITLE
the address dynamic arg for usecontracts

### DISCUSSION THREAD

**Discussion by zouhangwithsweet** [wevm/wagmi#711] [Q&A] [answered]
my token address is read from another abi，how can I use this hook in this case？

**Comment by tmm** [comment, marked as answer]
Check out the [`enabled`](https://wagmi.sh/docs/hooks/useToken#enabled-optional=) option. You can set that to `false` until your token address is loaded.

**Reply by zouhangwithsweet** [reply]
I got the error `contract.interface.encodeFunctionData is not a function`

<img width="719" alt="image" src="https://user-images.githubusercontent.com/21095710/179347306-edf325c8-9d55-4392-9c13-5acf711693dc.png">

**Reply by tmm** [reply]
Can you share a CodeSanbox? Hard to tell what's going on otherwise.

**Reply by zouhangwithsweet** [reply]
Sorry for the late reply. I had submited an issue #729

**Comment by zouhangwithsweet** [comment]
Example code

```ts
export const useTest = (tokenAddress: string) => {
  const auctionContract = useTestContract(tokenAddress)
  const { address } = useAccount()
  const { data } = useContractRead({
    ...auctionContract,
    functionName: 'startBlock',
    watch: true,
    // cacheOnBlock: true,
    // enabled: !!tokenAddress,
  })
console.log(data)
}

//
import { Interface } from '@ethersproject/abi'

export const useTestContract = (tokenAddress: string) => {
  return {
    addressOrName: tokenAddress,
    contractInterface: new Interface(TEST_ABI),
  }
}
```

**Reply by zouhangwithsweet** [reply]
with above error, the react-query will retry then error, back and forth cycle
